### PR TITLE
Docs: HTML form for getting in touch with Read the Docs for Science

### PR DESCRIPTION
**Pull request by benjaoming**
This is pending a Front endpoint for actually receiving form data.

image

---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--9543.org.readthedocs.build/en/9543/

- Developer's documentation (`dev`): https://dev--9543.org.readthedocs.build/en/9543/

### CODE REVIEW

**Comment by humitos** [comment]
I mentioned this quickly on Slack already. IMO, we are overloading the "Read the Docs Documentation" with things that are not a good fit as documentation resources. I'd prefer if we can put this form under https://github.com/readthedocs/site-community/ on the Science landing-page or any other page where you consider it fits better there.

Besides, I think that start doing the work on that repository will help us to speed things up and work towards shipping that site and making it online and available for readers 👍🏼

**Comment by benjaoming** [comment]
@humitos the destination of the Science page is also the site-community page, as a real landing page. See the comment at the top of science.rst

```
.. 2022-08-10
.. Notes about this section:
..
.. This section is intended as a "landing page", meaning that we will allow to
.. mix messages in a pragmatic way UNTIL a permanent location for this landing
.. page is found outside of the RTD User Documentation project.
.. more discussion: https://github.com/readthedocs/readthedocs.org/pull/9460/
..
.. The main ways that the page will be improved are:
.. - Add more focus to documentation perspectives
.. - Move all descriptions of "benefits", i.e. marketing to a separate location
.. - Likely stop using the dropdown element
.. - Continue to function as a "landing" page, but a landing page for
..   documentation resources for scientific/academic projects.
```

File: docs/user/science.rst [hunk 1]
```diff
@@ -159,3 +159,110 @@ Read the Docs community for science is already big and keeps growing. The :exter
...
+How would you use Read the Docs for Science?
+--------------------------------------------
+
+Would you like to get started with Read the Docs or understand more about the platform? Would you like to help us improve by telling us more about an already existing project?
+
+Please take 2 minutes to fill in this form.
+
+.. raw:: html
+
+    <form method="POST" action="fixme">
```
Comment (ericholscher) [inline docs/user/science.rst:173]:
```suggestion
<form
  method="POST"
  name="fa-form-1"
  action="https://webhook.frontapp.com/forms/036c4169294f3b04abaa/xP2Ulmxfcgl_mLJrFbGoefmVuqmH7DAfyHD9lt_qbk1heKFev5K8-TEhmpKc8dWdn-rv7bbZMMPjmffxl0mqGRUcrfyOzImtk8zEGJ04E1uuyPE28hqoHExtS20"
  enctype="multipart/form-data"
  accept-charset="utf-8"
>```

File: docs/user/science.rst [hunk 2]
```diff
@@ -159,3 +159,110 @@ Read the Docs community for science is already big and keeps growing. The :exter
...
+
+    <label for="what1">
+      <input id="what1" type="checkbox" name="interests-search-analytics" value="1">
+      Search and analytics
+    </label>
+
+    <label for="what1">
+      <input id="what1" type="checkbox" name="interests-search-analytics" value="1">
+      Previewing new proposals (pull requests)
+    </label>
```
Comment (ericholscher) [inline docs/user/science.rst:256]:
```suggestion
    </label>
    
    <input id="body" type="hidden" value="Science Docs Submission">
```

Front requires a `body` id.

**Review by ericholscher — APPROVED** [review]
Agreed this should go on a landing page, but this works for now.

Is there a reason we didn't just put the HTML in directly? Seems like an odd use of a table, but guessing it was for formatting?

**Comment by benjaoming** [comment]
@ericholscher the reason for the HTML table was to somehow use an expected layout and set of styling rules. I know it could have been done in a million ways, this was just the first idea that come on top of my head.

Gonna give it a test run now :)

**Comment by benjaoming** [comment]
Can errors be displayed on Front somehow so we don't have to handle them? I don't know if it's so important.

Currently on https://docs--9543.org.readthedocs.build/en/9543/science.html, I'm redirected to:

```
?code=bad_referer&message=This form was not submitted from the configured location.
```

I'll create a hidden branch build for docs.readthedocs.io to test this...

**Comment by benjaoming** [comment]
Answering my own message here: Nope, the error and success page is not that customizable, but we can write our own.

Is there a way to put an unreferenced .rst page in a Sphinx project without getting errors when building the project?

**Comment by benjaoming** [comment]
Form is tested and works. I think we can launch and consider error/success pages based on the first usage, :crossed_fingers: that someone will use it.

**Comment by humitos** [comment]
> Is there a way to put an unreferenced .rst page in a Sphinx project without getting errors when building the project?

I suppose what you are looking for is the metatag `:orphan:`.

```restructuredtext
.. science.rst
:orphan:

Title
===

The content of the file goes here
```

That tells Sphinx this page won't be referenced anywhere.

**Comment by ericholscher** [comment]
Agreed on shipping it with half-busted error pages. I'm guessing not many people will submit this form, so not too worried about it yet.

**Pull request merged**